probe-rs/src/architecture/riscv/mod.rs

```diff
@@ -706,16 +706,27 @@ impl<'probe> CoreInterface for Riscv32<'probe> {
         }
     }
 
-    fn fpu_support(&mut self) -> Result<bool, crate::error::Error> {
-        Err(crate::error::Error::Other(anyhow::anyhow!(
-            "Fpu detection not yet implemented"
-        )))
+    /// Returns the number of fpu registers defined in this register file, or `None` if there are none.
+    fn floating_point_register_count(&mut self) -> Result<Option<usize>, Error> {
+        let fpu_register_count = self
+            .registers()
+            .all_registers()
+            .filter(|r| r.register_has_role(crate::RegisterRole::FloatingPoint))
+            .count();
+        Ok(if fpu_register_count > 0 {
+            Some(fpu_register_count)
+        } else {
+            None
+        })
     }
 
-    fn floating_point_register_count(&mut self) -> Result<Option<usize>, crate::error::Error> {
-        Err(crate::error::Error::Other(anyhow::anyhow!(
-            "Fpu detection not yet implemented"
-        )))
+    fn fpu_support(&mut self) -> Result<bool, crate::error::Error> {
+        // Read the extensions from the Machine ISA regiseter.
+        let isa_extensions =
+            Misa::from(self.read_csr(Misa::get_mmio_address() as u16)?).extensions();
+        // Mask for the D, F and Q extension bits.
```

Comment:
Please add an explanation for this, it's not very clear what D, F, and Q mean here.